app/models/assignment.rb

```diff
@@ -904,7 +906,7 @@ def each_group_repo
   def self.get_repo_auth_records
     records = Assignment.joins(:assignment_properties)
                         .includes(groupings: [:group, { accepted_student_memberships: :user }])
-                        .where(assignment_properties: { vcs_submit: true })
+                        .where(is_hidden: false, assignment_properties: { vcs_submit: true })
```

Comment:
Great catch! The only problem is that we now hide assessments based on section membership as well. Implementing that is going to be tricky and I think it should actually be handled elsewhere. I can think about it and commit a change